libs/blocks/quiz/utils.js

```diff
@@ -92,20 +93,21 @@ export const handleResultFlow = async (answers = []) => {
   window.location.href = getRedirectUrl(destinationPage, primaryProductCodes, answers);
 };
 
-const storeResultInLocalStorage = (
+export const storeResultInLocalStorage = (
   answers,
   resultData,
   resultResources,
   primaryProducts,
   secondaryProductCodes,
   umbrellaProduct,
 ) => {
-  const nestedFrags = resultData.matchedResults[0]['nested-fragments'];
+  const nestedFragsPrimary = resultData.matchedResults[0]['nested-fragments-primary'];
+  const nestedFragsSecondary = resultData.matchedResults[0]['nested-fragments-secondary'];
   const structureFrags = resultData.matchedResults[0]['basic-fragments'];
 
   const structureFragsArray = structureFrags?.split(',');
-  const nestedFragsArray = nestedFrags?.split(',');
-
+  const nestedFragsPrimaryArray = nestedFragsPrimary?.split(',');
```

Comment:
When splitting, would it be possible to trim at the same time to ditch any whitespace? For example:
const nestedFragsPrimaryArray = nestedFragsPrimary?.split(',').map(item => item.trim());

Reply:
I love that idea of trimming when splitting itself. The only reason I did not go that route is to not have to iterate the `nestedFragsPrimaryArray` array twice. First with `.map(item => item.trim());` and second in `nestedFragments` function `nestedFragsPrimaryArray?.forEach((frag)`. While I know the size of the `nestedFragsPrimaryArray` will be small in most of the cases and the performance difference should be negligible.